### PR TITLE
Discuss the `ur` string prefix

### LINKED ISSUE #51: ur'strings' are OK in Python 2 but are syntax errors in Python 3
__python3 -c "ur'string'"__ --> Syntax Error

Some good advice could be distilled from:
* https://stackoverflow.com/questions/26063899/python-version-3-4-does-not-support-a-ur-prefix

### PR DESCRIPTION
**Pull request by encukou**
Also reword other parts of the strings section

Fixes: https://github.com/fedora-python/portingguide/issues/51

### CODE REVIEW

File: source/strings.rst [hunk 1]
```diff
@@ -161,13 +162,56 @@ Depending on what you need, explicitly use a serialization function
...
+    >>> print u"\x23☺\u2744"    # Python 2 with Encoding: UTF-8
+    #☺❄
+    >>> print ur"\x23☺\u2744"   # Python 2 with Encoding: UTF-8
+    \x23☺❄
+
+This may be confusing at first.
+Keeping this would be even more confusing in Python 3, where the ``u`` prefix
+is a no-op backwards-compatible behavior.
+Python 3 avoids the choice between confusing or backwards-incompatible
+semantics by forbidding ``ru`` altogether.
```
Comment (hroncok) [inline source/strings.rst]:
```suggestion
semantics by forbidding ``ur`` altogether.
```

File: source/strings.rst [hunk 2]
```diff
@@ -161,13 +162,56 @@ Depending on what you need, explicitly use a serialization function
...
+
+Avoid the ``ur`` prefixed in string literals.
```
Comment (hroncok) [inline source/strings.rst]:
```suggestion
Avoid the ``ur`` prefix in string literals.
```

**Review by hroncok — COMMENTED** [review]
Two suggestions. Yet I am not in the proper mindset to determine whether I like the suggested solution for raw unicode literals or not - such string literals seem horrid in both Pythons.

**Comment by encukou** [comment]
> Yet I am not in the proper mindset to determine whether I like the suggested solution for raw unicode literals or not - such string literals seem horrid in both Pythons.

You're right, I got carried away. A fully general algorithm that keeps the original literal would probably make a good fixer, but for humans let's just recommend doubling the backslash?

**Review by hroncok — APPROVED** [review]
Much better now.

File: source/strings.rst [hunk 3]
```diff
@@ -161,13 +162,51 @@ Depending on what you need, explicitly use a serialization function
...
+``\U........``), as the ``u`` prefix took precedence over ``r``::
+
+    >>> print u"\x23☺\u2744"    # Python 2 with Encoding: UTF-8
+    #☺❄
+    >>> print ur"\x23☺\u2744"   # Python 2 with Encoding: UTF-8
+    \x23☺❄
+
+This may be confusing at first.
+Keeping this would be even more confusing in Python 3, where the ``u`` prefix
+is a no-op backwards-compatible behavior.
```
Comment (encukou) [inline source/strings.rst]:
```suggestion
is a no-op with backwards-compatible behavior.
```

File: source/strings.rst [hunk 4]
```diff
@@ -161,13 +162,51 @@ Depending on what you need, explicitly use a serialization function
...
+is a no-op with backwards-compatible behavior.
+Python 3 avoids the choice between confusing or backwards-incompatible
+semantics by forbidding ``ur`` altogether.
+
+Avoid the ``ur`` prefix in string literals.
+
+The most straightforward way to do this is to use plain ``u`` literals
+with ``\\`` for a literal backslash::
+
+       >>> print(u"\\x23\u263a\u2744")
```
Comment (encukou) [inline source/strings.rst]:
```suggestion
       >>> print(u"\\x23☺\u2744")
```

**Comment by encukou** [comment]
Thanks!

**Pull request merged**